Mathlib/Data/Nat/Factorization/Basic.lean

```diff
@@ -651,6 +652,93 @@ theorem factorization_lcm {a b : ℕ} (ha : a ≠ 0) (hb : b ≠ 0) :
   exact (min_add_max _ _).symm
 #align nat.factorization_lcm Nat.factorization_lcm
 
+/-- If `a = ∏ pᵢ ^ nᵢ` and `b = ∏ pᵢ ^ mᵢ`, then `factorizationLCMLeft = ∏ pᵢ ^ kᵢ`, where
+`kᵢ = nᵢ` if `mᵢ ≤ nᵢ` and `0` otherwise. -/
+def factorizationLCMLeft (a b : ℕ) : ℕ :=
+  (Nat.lcm a b).factorization.prod fun p n ↦
+    if b.factorization p ≤ a.factorization p then p ^ n else 1
+
+/-- If `a = ∏ pᵢ ^ nᵢ` and `b = ∏ pᵢ ^ mᵢ`, then `factorizationLCMRight = ∏ pᵢ ^ kᵢ`, where
```

Comment:
If I'm wrong (I made the computation in my head), then it is even more important to add a comment — and why not prove the thing, you more or less use it later on in the proofs, I think.

Reply:
If `b = 0`, all of its exponents are `0`, so you get the same thing as the exponents of `a`, which is `a` if `a>0`, and `1` otherwise.
If `a = 0`, then all `k_i` are `0`, so you get 1.
In other words, for `(0,b)`it gives `1`, and for `(a,0)` with `a>0`, it gives `a`.
Seeing what you copied, you may have used `factorizationLCMRight`.

Reply:
The product is over the divisors of `lcm a b`, so if one of `a` or `b` is `0` then the result is `1` (I've added the results).